### PR TITLE
Add new tutorial on upgrading

### PR DESCRIPTION
**Pull request by richlander**
We are about to publish a blog post on .NET 7 EOL. Past posts like that have too much content and should primarily rely on content in docs.

Example:
https://devblogs.microsoft.com/dotnet/dotnet-5-end-of-support-update/

The tutorial is intended to be that content. I didn't find a doc like this. If there is, we can use that instead. 

Is this a good home?

---

#### Internal previews

| 📄 File | 🔗 Preview link |
|:--|:--|
| [docs/core/install/upgrade.md](https://github.com/dotnet/docs/blob/4f5d62cdd20273c0ecc0d939c1892bd9a946c9b7/docs/core/install/upgrade.md) | [Upgrade to a new .NET version](https://review.learn.microsoft.com/en-us/dotnet/core/install/upgrade?branch=pr-en-us-40212) |

### CODE REVIEW

**Comment by richlander** [comment]
I noticed a special mixture of "updating" and "upgrading". Well done!

File: docs/core/tutorials/upgrade.md [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+- [Linux](https://learn.microsoft.com/dotnet/core/install/linux)
+- [Windows](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/DotNet/SDK)
+
+Visual Studio installs [new .NET SDK versions automatically](https://learn.microsoft.com/dotnet/core/porting/versioning-sdk-msbuild-vs). For Visual Studio users, upgrading to a newer Visual Studio version will be sufficient.
+
+## Upgrading source code
+
+The only required change to upgrade an app is updating the `TargetFramework` property in a project file to the newer .NET version.
+
+Here's how to do it:
```
Comment (timheuer) [inline docs/core/tutorials/upgrade.md:39]: We have tooling for upgrades, we should absolutely be mentioning them https://aka.ms/dotnet-upgrade-assistant -- it helps automate all of this for you and give you a nice report of more tasks you might need to do!

File: docs/core/tutorials/upgrade.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+## Requesting an upgrade (as a user)
+
+You may need an app to use a new .NET version and to make a rquest to the team or vendor that produced the app.
+
+Common reasons to upgrade to a new .NET version:
+
+- The currently used .NET version is no longer supported
+- The new version supports a new operating system
+- The new version has an important performance or security feature
```
Comment (agocke) [inline docs/core/tutorials/upgrade.md:19]: Also has a new API or "feature" in general?

**Review by agocke — APPROVED** [review]

File: docs/core/tutorials/upgrade.md (hunk 2 above)
Comment (richlander) [reply docs/core/tutorials/upgrade.md:19]: I was writing that from the point of an end-user. They are less likely to know about an API. Perhaps I should just delete/re-frame that whole section and then include a longer description of "why upgrade". What do you think?

File: docs/core/tutorials/upgrade.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+title: Upgrade to a new .NET version
+description: Learn how to upgrade an app to a new .NET version.
+ms.topic: how-to
+ms.date: 03/25/2024
+---
+# Upgrade to a new .NET version
+
+New .NET versions are [released each year](https://github.com/dotnet/core/blob/main/releases.md). Many developers start the upgrade process as soon as the new version is available, while others wait until the version they are using is no longer supported. The upgrade process has multiple aspects to consider.
+
+## Requesting an upgrade (as a user)
```
Comment (gewarren) [inline docs/core/tutorials/upgrade.md:11]:
```suggestion
## Request an upgrade (as a user)
```

Comment (gewarren) [reply docs/core/tutorials/upgrade.md:11]: We're supposed to avoid -ing words per MS style guide.

File: docs/core/tutorials/upgrade.md [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+You may need an app to use a new .NET version and to make a rquest to the team or vendor that produced the app.
```
Comment (gewarren) [inline docs/core/tutorials/upgrade.md:13]:
```suggestion
You might need an app that you use to be upgraded to use a new .NET version. In this case, make a request to the app developer.
```

File: docs/core/tutorials/upgrade.md [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+
+Common reasons to upgrade to a new .NET version:
+
+- The currently used .NET version is no longer supported
+- The new version supports a new operating system
+- The new version has an important performance or security feature
+
+## Upgrading development environment
```
Comment (gewarren) [inline docs/core/tutorials/upgrade.md:21]:
```suggestion
## Upgrade development environment
```

File: docs/core/tutorials/upgrade.md [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+
+Some operating systems have a package manager that can be used to install a new .NET version, which may be a prefered option for some users.
+
+- [macOS](https://formulae.brew.sh/cask/dotnet-sdk)
+- [Linux](https://learn.microsoft.com/dotnet/core/install/linux)
+- [Windows](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/DotNet/SDK)
+
+Visual Studio installs [new .NET SDK versions automatically](https://learn.microsoft.com/dotnet/core/porting/versioning-sdk-msbuild-vs). For Visual Studio users, upgrading to a newer Visual Studio version will be sufficient.
+
+## Upgrading source code
```
Comment (gewarren) [inline docs/core/tutorials/upgrade.md:35]:
```suggestion
## Upgrade source code
```

File: docs/core/tutorials/upgrade.md (hunk 1 above)
Comment (gewarren) [reply docs/core/tutorials/upgrade.md:39]: You should probably link here instead: https://learn.microsoft.com/en-us/dotnet/core/porting/upgrade-assistant-overview (../porting/upgrade-assistant-overview.md)

File: docs/core/tutorials/upgrade.md [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+
+`dotnet workload restore` may be needed to restore workloads with the new SDK version.
+
+More resources:
+
+* [Breaking changes in .NET 8](https://docs.microsoft.com/dotnet/core/compatibility/8.0)
+* [Migrate from ASP.NET Core in .NET 7 to .NET 8](https://learn.microsoft.com/aspnet/core/migration/70-80?view=aspnetcore-8.0&tabs=visual-studio)
+* [Upgrading .NET MAUI from .NET 7 to .NET 8](https://github.com/dotnet/maui/wiki/Upgrading-.NET-MAUI-from-.NET-7-to-.NET-8)
+
+## Updating continuous integration (CI)
```
Comment (gewarren) [inline docs/core/tutorials/upgrade.md:55]:
```suggestion
## Update continuous integration (CI)
```

File: docs/core/tutorials/upgrade.md [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+
+CI pipelines follow a similar update process as project files and Dockerfiles. [CI pipelines](https://github.com/actions/setup-dotnet) can typically be updated with changing only version values.
+
+
+## Updating hosting environment
```
Comment (gewarren) [inline docs/core/tutorials/upgrade.md:60]:
```suggestion
## Update hosting environment
```

**Review by gewarren — CHANGES_REQUESTED** [review]
Thanks for adding this. I think it probably belongs in this section of the TOC/docs - [Install .NET on Windows, Linux, and macOS | Microsoft Learn](https://learn.microsoft.com/en-us/dotnet/core/install/). It shouldn't be in the tutorials folder since it's not a tutorial. Specifically, it should probably live just above this article in the TOC:

image

**Comment by richlander** [comment]
@gewarren -- I thought about putting this doc in that exact spot. However, I hesitated. It's not really about install as an intent, but upgrading to a new version. Perhaps that's splitting hairs. I'm happy to put it there if that's what folks think is best.

It's really "getting started with a new .NET version".

**Comment by tdykstra** [comment]
> @gewarren -- I thought about putting this doc in that exact spot. However, I hesitated. It's not really about install as an intent, but upgrading to a new version. Perhaps that's splitting hairs. I'm happy to put it there if that's what folks think is best.
> 
> It's really "getting started with a new .NET version".

I agree the location proposed by @gewarren isn't a perfect fit, but I think it's the best we have in this doc set. We have a whole TOC section devoted to [upgrades in the ASP.NET Core docs](https://learn.microsoft.com/en-us/aspnet/core/migration/70-80?view=aspnetcore-8.0).

**Comment by richlander** [comment]
I believe I have resolved the feedback.

Mega thanks for the help and to @gewarren for all the edits.

**Review by jongalloway — APPROVED** [review]

**Review by gewarren — APPROVED** [review]

File: docs/core/install/upgrade.md [hunk 9]
```diff
@@ -0,0 +1,69 @@
...
+## Upgrade development environment
+
+To upgrade to a new .NET version, the .NET SDK is the primary component to install. It includes an updated .NET CLI, build system, and runtime version.
+
+The .NET website offers [installers and archives](https://dotnet.microsoft.com/download) that you can download and use on any supported operating system and architecture.
+
+Some operating systems have a package manager that you can also use to install a new .NET version, which you might prefer.
+
+- [macOS](https://formulae.brew.sh/cask/dotnet-sdk)
+- [Linux](https://learn.microsoft.com/dotnet/core/install/linux)
```
Comment (gewarren) [inline docs/core/install/upgrade.md:26]:
```suggestion
- [Linux](linux.md)
```

File: docs/core/install/upgrade.md [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+- [Windows](https://github.com/microsoft/winget-pkgs/tree/master/manifests/m/Microsoft/DotNet/SDK)
+
+Visual Studio installs [new .NET SDK versions automatically](https://learn.microsoft.com/dotnet/core/porting/versioning-sdk-msbuild-vs). For Visual Studio users, it's sufficient to upgrade to a newer Visual Studio version.
```
Comment (gewarren) [inline docs/core/install/upgrade.md:29]:
```suggestion
Visual Studio installs [new .NET SDK versions automatically](../porting/versioning-sdk-msbuild-vs.md). For Visual Studio users, it's sufficient to upgrade to a newer Visual Studio version.
```

File: docs/core/install/upgrade.md [hunk 11]
```diff
@@ -0,0 +1,69 @@
...
+
+For containers, [`FROM` statements](https://github.com/dotnet/dotnet-docker/blob/e5e8164460037e77902cd269c788eccbdeea5edd/samples/aspnetapp/Dockerfile#L17) need to be changed to include new version numbers.
+
+The following Dockerfile example demonstrates pulling an ASP.NET Core 8.0 image.
+
+```dockerfile=
+FROM mcr.microsoft.com/dotnet/aspnet:8.0
+```
+
+In a cloud service like [Azure App Service](https://learn.microsoft.com/en-us/azure/app-service/quickstart-dotnetcore), a configuration change is needed.
```
Comment (gewarren) [inline docs/core/install/upgrade.md:69]:
```suggestion
In a cloud service like [Azure App Service](/azure/app-service/quickstart-dotnetcore), a configuration change is needed.
```

**Review by gewarren — APPROVED** [review]

**Pull request merged**